src/resources/views/crud/fields/checklist.blade.php

```diff
@@ -89,8 +97,38 @@ function bpFieldInitChecklist(element) {
 
                   hidden_input.val(JSON.stringify(newValue));
 
+                  toggleAllSelectCheckbox();
                 });
             }
+
+            document.getElementById("select-all").addEventListener("click", selectAll);
+
+            function selectAll() {
+                var allCheckbox = $('#checkbox-wrapper :input[type=checkbox]');
+
+                if ($("#select-all").is(':checked')) {
+                  allCheckbox.each(function(){
+                    if ($(this).is(':checked') == false) {
+                      $(this).click();
+                    }
+                  });
+                } else {
+                  allCheckbox.each(function(){
+                    $(this).click();
+                  });
+                }
+            };
+
+            function toggleAllSelectCheckbox() {
+              var selectedItems = JSON.parse(document.getElementsByName("{{ $field['name'] }}")['0'].value);
+              var allItems = "{{ count($field['options']) }}";
+
+              if(selectedItems.length == allItems) {
+                $('#select-all').prop('checked', 'checked');
+              } else {
+                $('#select-all').prop('checked', false);
+              }
+            }
```

Comment:
This would also not be needed if it were an anchor... right?

Reply:
This is probably going away, so no use in changing. But I have a hot tip for you regarding JS inside Backpack fields: try to avoid Blade echos inside JS. Whenever I see stuff like 
```js
var allItems = "{{ count($field['options']) }}";
```

I think to myself... ok so this will make my JS non-reusable. How can I avoid this? And the answer, most of the time, is using `data-attributes`. I put this bit of logic in HTML, instead of JS. Something like:

```php
<div class="items" data-item-count="{{ count($field['options']) }}"></div>

// then in JS
var allItems = $('.items').data('item-count');
```

It's a small change, but makes the JS completely independent from the variables.

Not important here if this goes away, but I just wanted to share how we normally do things in Backpack fields, maybe it helps you in other fields.